core/src/main/scala/kafka/tools/ConsoleProducer.scala

```diff
@@ -302,22 +304,32 @@ object ConsoleProducer {
       if (props.containsKey("key.separator"))
         keySeparator = props.getProperty("key.separator")
       if (props.containsKey("parse.headers"))
-        parseHeader = props.getProperty("parse.headers").trim.equalsIgnoreCase("true")
+        parseHeaders = props.getProperty("parse.headers").trim.equalsIgnoreCase("true")
       if (props.containsKey("headers.delimiter"))
         headersDelimiter = props.getProperty("headers.delimiter")
       if (props.containsKey("headers.separator"))
         headersSeparator = props.getProperty("headers.separator")
-        headerSeparatorPattern = Pattern.compile(headersSeparator)
+        headersSeparatorPattern = Pattern.compile(headersSeparator)
```

Comment:
nit: Not related to you change. The indentation of this line is weird. Should we align it on the `if` instead?